File: Leetcode/Math/633. Sum of Square Numbers.py
from math import sqrt

class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        for i in range(int(sqrt(c)) + 1):
            jj = c - i*i
            j = int(sqrt(jj))

            if j*j == jj and (i*i + j*j) == c:
                return True
            
        return False
        

File: Leetcode/Practice/Math/633. Sum of Square Numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        for(int i = 0; (long long) i*i <= c; i++) {
            int jj = c - i*i;
            int j = sqrt(jj);
            if(j*j == jj) return true;
        }
        return false;
    }
};